avl/tree.c

```diff
@@ -0,0 +1,274 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <math.h>
+#include "tree.h"
+
+typedef struct Node {
+    char* key;
+    char* value;
+    int height;
+    struct Node* left;
+    struct Node* right;
+} Node;
+
+struct Dictionary {
+    Node* root;
+};
+
+Dictionary* createDictionary() {
+    Dictionary* tree = calloc(1, sizeof(Dictionary));
+    if (tree == NULL) {
+        return NULL;
+    }
+    tree->root = NULL;
+    return tree;

```

Comment:
```suggestion
    return calloc(1, sizeof(Dictionary));
```
Это абсолютно одно и то же :)